src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -500,3 +504,584 @@ def _for_interval_sample_size_calculation(i):
+
+    Currently, only Firefox Desktop is supported.
+
+    experiment_name: name slug for the analysis, used in BQ table names
+    bq_context: BigQuery configuration and client
+    metrics: list of mozanalysis Metrics to analyze
+    targets: list mozanalysis Segments defining user population to include. For sizing,
+        only clients in the intersection of all targets are included.
+    n_days_observation_max: max number of days the observation period will cover.
+        The historical period will be chosen to allow for this observation length,
+        in case sizing is computed for different time periods.
+    n_days_enrollment: number of days over which clients are enrolled
+    n_days_launch_after_version_release: number of days after version release to begin enrollment
+    end_date: most recent date the historical period should cover. Set this to push
+        the historical period further into the past. If unspecified, will be set to "yesterday".
+    alpha: desired significance level of the experiment
+    power: desired power of the experiment
+    outlier_percentile: metric values higher than this percentile level will be trimmed.
+        Specified as a number between 0 and 1. Set to 0 to disable.
+    sample_rate: sampling rate to apply to the population, specified as a number between 0 and 1.
+        If set, only this proportion of clients will be retained from the targets
+        for sizing calculations. Set to 0 to disable.
+    """
+
+    def __init__(
+        self,
+        experiment_name: str,
+        bq_context: BigQueryContext,
+        metrics: List[Metric],
+        targets: List[Segment],
+        n_days_observation_max: int,
+        n_days_enrollment: int = 7,
+        n_days_launch_after_version_release: int = 7,
+        end_date: Optional[str] = None,
+        alpha: float = 0.05,
+        power: float = 0.9,
+        outlier_percentile: float = 0.995,
+        sample_rate: float = 0.01,
+    ):
+        self.bq_context = bq_context
+        self.experiment_name = experiment_name
+        self.metrics = list(metrics)
+        self.targets = list(targets)
+
+        self.n_days_observation_max = n_days_observation_max
+        self.n_days_enrollment = n_days_enrollment
+        self.n_days_launch_after_version_release = n_days_launch_after_version_release
+        self.end_date = end_date
+
+        self.alpha = alpha
+        self.power = power
+        self.outlier_percentile = outlier_percentile
+        self.sample_rate = 0 if sample_rate >= 1 else sample_rate
+
+        self.total_historical_period_days = (
+            n_days_launch_after_version_release
+            + n_days_enrollment
+            + n_days_observation_max
+        )
+        # Will be set after pulling data
+        self._population_size = None
+
+        # Sets self.min_major_version, self.min_major_version_date
+        self._find_latest_feasible_historical_version()
```

Comment:
As we discussed, it'd be nice to have this as optional, for experiments where we aren't targeting a recent release.